### PR TITLE
Add FreeBSD support

### PR DESCRIPTION
**Pull request by hajimehoshi**
This PR adds FreeBSD support. See also https://github.com/go-gl/gl/issues/74.

### CODE REVIEW

File: tmpl/package.tmpl [hunk 1]
```diff
@@ -120,10 +120,10 @@ func {{.GoName}}({{template "paramsGoDecl" .Parameters}}){{if not .Return.IsVoid
 // again after switching between OpenGL contexts may work if the contexts belong
 // to the same graphics driver/device).
 //
-// On OS X and Linux, the behavior is different, but code written compatible
-// with the Windows behavior is compatible with OS X and Linux. That is, always
-// Init under an active OpenGL context, and always re-init after switching
-// graphics contexts.
+// On macOS and the other POSIX systems, the behavior is different, but code
+// written compatible with the Windows behavior is compatible with macOS and the
+// other POSIX systems. That is, always  Init under an active OpenGL context, and
```
Comment (shurcooL) [inline tmpl/package.tmpl]: Two spaces between "always" and "Init", should be one space.

File: tmpl/package.tmpl [hunk 2]
```diff
@@ -24,9 +24,9 @@ package {{.Name}}
 {{define "paramsGoDecl"}}{{range $i, $p := .}}{{if ne $i 0}}, {{end}}{{$p.GoName}} {{$p.Type.GoType}}{{end}}{{end}}
 {{define "paramsGoCall"}}{{range $i, $p := .}}{{if ne $i 0}}, {{end}}{{$p.Type.ConvertGoToC $p.GoName}}{{end}}{{end}}
 
-// #cgo darwin  LDFLAGS: -framework OpenGL
-// #cgo linux   LDFLAGS: -lGL
-// #cgo windows LDFLAGS: -lopengl32
+// #cgo darwin        LDFLAGS: -framework OpenGL
+// #cgo freebsd linux LDFLAGS: -lGL
```
Comment (shurcooL) [inline tmpl/package.tmpl]: Can you reorder them so linux comes before freebsd? Linux is more popular and well supported.

File: tmpl/procaddr.tmpl [hunk 3]
```diff
@@ -3,7 +3,7 @@
 //
 // windows: WGL
 // darwin: CGL
-// linux: GLX
+// (the other POSIX): GLX
```
Comment (shurcooL) [inline tmpl/procaddr.tmpl]: Please list them explicitly, it's only 2 builds tags. It'll be more clear in the future.

File: tmpl/procaddr.tmpl [hunk 4]
```diff
@@ -20,8 +20,8 @@ package {{.Name}}
 #cgo darwin CFLAGS: -DTAG_DARWIN
 #cgo darwin LDFLAGS: -framework OpenGL
 
-#cgo linux CFLAGS: -DTAG_LINUX
-#cgo linux LDFLAGS: -lGL
+#cgo freebsd linux CFLAGS: -DTAG_POSIX
+#cgo freebsd linux LDFLAGS: -lGL
```
Comment (errcw) [inline tmpl/procaddr.tmpl]: same ordering nit as above, please list linux first

File: tmpl/package.tmpl (hunk 2 above)
Comment (hajimehoshi) [inline tmpl/package.tmpl]: I tried to use alphabetical order as the standard lib does. I'm fine with either way.

**Comment by hajimehoshi** [comment]
> Have you tested the results of this change on Linux and FreeBSD?

I have tried only on FreeBSD and now this succeeded after your reverting a problematic commit.

File: tmpl/package.tmpl (hunk 2 above)
Comment (hajimehoshi) [inline tmpl/package.tmpl]: Done

File: tmpl/package.tmpl (hunk 1 above)
Comment (hajimehoshi) [inline tmpl/package.tmpl]: Done.

File: tmpl/procaddr.tmpl (hunk 3 above)
Comment (hajimehoshi) [inline tmpl/procaddr.tmpl]: Done.

File: tmpl/procaddr.tmpl (hunk 4 above)
Comment (hajimehoshi) [inline tmpl/procaddr.tmpl]: Done.

File: tmpl/package.tmpl (hunk 2 above)
Comment (hajimehoshi) [inline tmpl/package.tmpl]: Done.

**Comment by hajimehoshi** [comment]
I'm fine with doing that tonight. BTW, why don't you use TravisCI for such testing?

**Comment by errcw** [comment]
LGTM too.

We have Travis set up for go-gl/gl, where the actual generated sources are, but not for gl-gl/glow. You're right that it's likely worth including a test case in glow so we discover problems before we try and generate code for gl.

**Comment by hajimehoshi** [comment]
@shurcooL I've confirmed that my PR worked on Linux!

**Pull request merged**